The job **`test-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/8316618467/job/22756379882) [(plain)](https://github.com/rust-lang-ci/rust/commit/62beff8e5ce08b52c342f716abbb9530ad924d89/checks/22756379882/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling core v0.0.0 (/checkout/library/core)
[RUSTC-TIMING] corebenches test:true 0.049
[RUSTC-TIMING] coretests test:true 41.168
    Finished release [optimized] target(s) in 42.09s
     Running tests/lib.rs (obj/build/x86_64-unknown-linux-gnu/stage1-std/wasm32-wasip1/release/deps/coretests-1863138ae991d121.wasm)
error: the option `Z` is only accepted on the nightly compiler
error: test failed, to rerun pass `-p core --test coretests`
  local time: Sun Mar 17 16:32:23 UTC 2024
  network time: Sun, 17 Mar 2024 16:32:23 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/bootstrap/src/lib.rs
+++ b/src/bootstrap/src/lib.rs
@@ -1387,6 +1387,9 @@ impl Build {
         if let Some(path) = finder.maybe_have("wasmtime") {
             if let Ok(mut path) = path.into_os_string().into_string() {
                 path.push_str(" run -C cache=n --dir .");
+                // Make sure that tests have access to RUSTC_BOOTSTRAP. This (for example) is
+                // required for libtest to work on beta/stable channels.
+                path.push_str(" --env RUSTC_BOOTSTRAP");
                 return Some(path);
             }
         }

--- a/tests/ui/feature-gates/test-listing-format-json.rs
+++ b/tests/ui/feature-gates/test-listing-format-json.rs
@@ -5,6 +5,7 @@
 //@ check-run-results
 //@ ignore-nightly
 //@ unset-exec-env:RUSTC_BOOTSTRAP
+//@ ignore-wasi cannot unset RUSTC_BOOTSTRAP
 
 #![cfg(test)]
 #[test]

